optimum/intel/openvino/modeling_sentence_transformers.py

```diff
@@ -0,0 +1,180 @@
+
+        inputs = {
+            "input_ids": input_ids,
+            "attention_mask": attention_mask,
+        }
+
+        # Add the token_type_ids when needed
+        if "token_type_ids" in self.input_names:
+            inputs["token_type_ids"] = token_type_ids if token_type_ids is not None else np.zeros_like(input_ids)
+
+        outputs = self._inference(inputs)
+        return {
+            "token_embeddings": torch.from_numpy(outputs["token_embeddings"]).to(self.device),
+            "sentence_embedding": torch.from_numpy(outputs["sentence_embedding"]).to(self.device),
+        }
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        load_in_8bit: Optional[bool] = None,
+        quantization_config: Union[OVWeightQuantizationConfig, Dict] = None,
+        **kwargs,
+    ):
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+        # This attribute is needed to keep one reference on the temporary directory, since garbage collecting
+        # would end-up removing the directory containing the underlying OpenVINO model
+        cls._model_save_dir_tempdirectory_instance = save_dir
+
+        # If load_in_8bit and quantization_config not specified then ov_config is set to None and will be set by default in convert depending on the model size
+        if load_in_8bit is None and not quantization_config:
+            ov_config = None
+        else:
+            ov_config = OVConfig(dtype="fp32")
+
+        # OVModelForFeatureExtraction works with Transformers type of models, thus even sentence-transformers models are loaded as such.
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            task=task or cls.export_feature,
+            subfolder=subfolder,
+            revision=revision,
+            cache_dir=cache_dir,
+            token=token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+            ov_config=ov_config,
+            library_name="sentence_transformers",
```

Comment:
If `library_name` is not specified we use transformer pipeline for compatibility: https://github.com/huggingface/optimum-intel/blob/main/optimum/exporters/openvino/__main__.py#L239.

Reply:
> If library_name is not specified we use transformer pipeline for compatibility: https://github.com/huggingface/optimum-intel/blob/main/optimum/exporters/openvino/__main__.py#L239.

Yes, I think we could remove this now that we have a `OVModelForSentenceTransformer` class to load the resulting models, even though we need to be careful as this is a breaking change, cc @eaidova to also have your opinion on this

Reply:
can we remove this behvaiour later (on next release)? there are embeddings models integrations in some libs that expects models in transformers format only and it will requires some time to switch them to support models converted as sentence transformers (or at least updating recommendation to explicitly set --library for this case)